docs/guides/guide-for-writing-plugins.md

```diff
@@ -0,0 +1,71 @@
+---
+description: |
+  You need to use external code for performance 
+  reason or to avoid re inventing the wheel? This guide
+  explains how to integrate your code as a plugin.
+---
+
+# Guide for plugin development
+
+This page helps you get started building plugins for typst.
```

Comment:
I think it needs to be made more explicit that this guide is for more "advanced" users (regular users probably won't know what WebAssembly is and stuff, and will arrive at this page very confused). Maybe some "(Advanced)" sign could be added next to the title, along with a more explicit disclaimer at the beginning like "This guide is intended for developers who wish to create advanced plugins for Typst using the WebAssembly platform." Not sure of the best wording or text structure here, though.

Reply:
feel free to tell me what you think of the new wording!